### PR TITLE
Remove SpectrumFit _est_errors_iminuit ?

### PR DESCRIPTION
**Pull request by cdeil**
@joleroi - I think today we said that [SpectrumFit._est_errors_iminuit](https://github.com/gammapy/gammapy/blob/57363f840a11826f1e26187055d3bc360d3e93b5/gammapy/spectrum/fit.py#L427) can be removed, now that we have [this](https://github.com/gammapy/gammapy/blob/57363f840a11826f1e26187055d3bc360d3e93b5/gammapy/utils/fitting/iminuit.py#L54)?

Can the whole `covar_axis` and `covariance` attribute be removed, now that it's on `ParameterList`?

Can you please make sure there's a test covering this, and then do this cleanup?

### CODE REVIEW

**Comment by cdeil** [comment]
Done in the commits attached here. Tests pass locally. Merging.

File: gammapy/spectrum/fit.py [hunk 1]
```diff
@@ -414,40 +406,8 @@ def _make_fit_result(self, parameters):
...
-            iminuit_covariance = self._iminuit_fit.covariance
-            cov = np.array([iminuit_covariance[c] for c in parameter_combinations])
-        else:
-            # fit did not converge
-            cov = np.repeat(np.nan, len(parameter_combinations))
-
-        cov = cov.reshape(len(parameter_names), -1)
-        self.covariance = cov
+        # TODO: add this back once fitting backends support optimisation
+        # and error estimation as separate steps
```
Comment (joleroi) [inline gammapy/spectrum/fit.py]: Are you sure this works? I would have expected a ``pass`` here ...

Comment (cdeil) [reply gammapy/spectrum/fit.py]:
```
>>> def spam():
...     """It works if there's a docstring!"""
... 
>>>
```

Comment (joleroi) [reply gammapy/spectrum/fit.py]: Always write docstrings :point_up:

**Pull request merged**